lib/AST/ASTContext.cpp

```diff
@@ -1053,9 +1056,9 @@ ASTContext::getBuiltinInitDecl(NominalTypeDecl *decl,
   return witness;
 }
 
-FuncDecl *ASTContext::getEqualIntDecl() const {
-  if (getImpl().EqualIntDecl)
-    return getImpl().EqualIntDecl;
+FuncDecl *ASTContext::getBinaryComparisonOperatorIntDecl(StringRef op, FuncDecl **cached) const {
```

Comment:
Nit: Wary of this being a double pointer. Can you take reference-to-pointer here instead?

Reply:
Also, there's no reason this needs to be exposed API.  Could you make it `static`?

Reply:
it requires access to `getIntDecl` and `getBoolDecl`

Reply:
It can take a reference to the ASTContext then.

Reply:
done